06-building-a-program-start-to-finish/writing-tests/output/invertedIndex-spec.js

```diff
@@ -0,0 +1,205 @@
+const invalid = require('./allBooks/in-valid.json');
+// empty array
+const emptyFile = require('./allBooks/emptyArray.json');
+// an invalid JSON file
+const notValid = require('./allBooks/notArrayOfArray.json');
+
+describe('invertedIndex Index', () => {
+   beforeAll(() => {
+    this.invertedIndex = new InvertedIndex();
+  });
+  
+  describe('Constructor', () => {
+    it('can create instances of inverted index class', () => {
+      expect(typeof this.invertedIndex)
+        .toEqual('object');
+      expect(this.invertedIndex instanceof InvertedIndex)
+        .toBe(true);
+    });
+  });
+
+  const invertedIndex = new InvertedIndex();
+  invertedIndex.createIndex('book.json', book);
+  invertedIndex.createIndex('secondBook.json', secondBook);
+
+  describe('InvertedIndex class, check all methods', () => {
+    it('should check that the class has a createIndex method', () => {
+      expect(typeof invertedIndex.createIndex).toBe('function');
+    });
+
+    it('should check that the class has a validateFile method', () => {
+      expect(typeof InvertedIndex.validateFile).toBe('function');
+    });
+
+    it('should check that the class has a tokenize method', () => {
+      expect(typeof InvertedIndex.tokenize).toBe('function');
+    });
+
+    it('should check that the class has a getIndex method', () => {
+      expect(typeof invertedIndex.getIndex).toBe('function');
+    });
+
+    it('should check that the class has a searchIndex method', () => {
+      expect(typeof invertedIndex.searchIndex).toBe('function');
+    });
+
+    it('should check that the class has a setIndex method', () => {
+      expect(typeof invertedIndex.setIndex).toBe('function');
+    });
+  });
+
+  describe('validateFile should check files', () => {
+    it('should check that the contents of the file to be uploaded is valid',
+    () => {
+      expect(InvertedIndex.validateFile(book)).toBeTruthy();
+    });
+
+    it('should return false for empty json files', () => {
+      expect(InvertedIndex.validateFile(empty)).toBeFalsy();
+    });
+
+    it('should return true if file has property "title" and "text" ', () => {
+      expect(InvertedIndex.validateFile(book)).toBeTruthy();
+    });
+
+    it('should return false if file does not have property "title" and "text"',
+     () => {
+       expect(InvertedIndex.validateFile(invalidFile)).toBeFalsy();
+     });
+
+    it('should return false if file is not an array of JSON object',
+     () => {
+       expect(InvertedIndex.validateFile(invalid)).toBeFalsy();
+     });
+
+    it('should return false if file is an empty array',
+     () => {
+       expect(InvertedIndex.validateFile(emptyFile)).toBeFalsy();
+     });
+
+    it('should return false if file is a JSON file but not' +
+    'an array of an array',
+     () => {
+       expect(InvertedIndex.validateFile(notValid)).toBeFalsy();
+     });
+  });
+
+  describe('Create Index', () => {
+    it('should return a msg if index is not created', () => {
```

Comment:
What about when the index is created?

Don't we also need to test that the index was created correctly?